src/main/resources/org/sonar/plugins/checkstyle/rules.xml

```diff
@@ -814,9 +814,6 @@
     <tag>comment</tag>
     <configKey><![CDATA[Checker/TreeWalker/JavadocMethod]]></configKey>
     <cardinality>MULTIPLE</cardinality>
-    <param key="minLineCount" type="INTEGER">
-      <defaultValue>-1</defaultValue>
-    </param>
```

Comment:
we need to update issue description with migration notes to let user know what to do when problem happen with Check after upgrade.

Please configure JavadocMethodCheck on already released version of plugin, run it, upgrade to new version of plugin, run again, experience problem, and we need to explain users how resolve it. I think user will have to recreate Rule from scratch.

please share exception in issue, to let users quickly find migration notes by such exception.

Reply:
I thought already this won't fail. The user can't configure checks with the old/deleted properties but current checks won't fail because the underlying CS still accpets those properties.
Interestingly, which I wasn't aware of, SQ seems to save a copy of the old template for every check, meaning I can actually reconfigure my currently existing check with all the old properties but just can't create a new one using those removed properties.
However, I also couldn't get this old check raising any violations in my example code although with the previous SQ plugin version I have got some.
How should we declare and resolve this "issue"?

Reply:
> because the underlying CS still accpets those properties.

oh , yes, such removal was non standard, as original Check was not changed. Yes, it will continue to work till we actually remove properties from Check class.

> However, I also couldn't get this old check raising any violations in my example code although with the previous SQ plugin version I have got some.

Do you mean:
You create configure JavadocMethod on 8.20, and it works, show some violation.
After upgrade to 8.21, it stops reporting violations on the same places ?

Its is probably because we removed from Check all functionality that was moved to another Check - https://github.com/checkstyle/checkstyle/commit/f094c67d6c042a0f786665034b5dddbb665b955c#diff-01491bb8516b661ce590f3bc79015eedR173
looks like `setAllowMissingParamTags` still in this Check, try to play with it in confg , it should still be in action.

Reply:
Yes, exactly this is what I have observed. With 8.20 it reported violations and with 8.21 none were reported and your explanation makes totally sense. If the functionality is removed but all properties still "there" (but deprecated) it won't fail but also not report anything anymore.

Reply:
Just put in mutation notes, most users are not aware that sonar plugin and checkstyle are different projects

Reply:
Perfect, I just created a migration comment: https://github.com/checkstyle/sonar-checkstyle/issues/222#issuecomment-503264472